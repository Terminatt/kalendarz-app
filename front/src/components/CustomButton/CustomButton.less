@import '~styles/variables.less';
@import '~styles/mixins.less';

.ant-btn.custom-btn {
  background-color: transparent;
  padding: @padding-tiny @padding-medium;
  height: max-content;
  border-radius: @border-radius-regular;
  font-weight: bold;
  
  &-primary {
    border: 2px solid @primary-color;
  }

  &-default {
    border: 2px solid @border-color-base;
  }

  &-default:hover {
    background-color: transparent;
    border-color: darken(@border-color-base, 15%);
  }

  &-primary:hover {
    background-color: transparent;
    border-color: darken(@primary-color, 15%);
  }

  &-clear {
    border: 2px solid @warning-color;
  }

  &-clear:hover {
    background-color: transparent;
    border-color: darken(desaturate(@warning-color, 25%), 15%);
  }

  &-delete {
    border: 2px solid @red-color;
  }

  &-delete:hover {
    background-color: transparent;
    border-color: darken(@red-color, 15%);
  }

  &:hover {
    background-color: transparent;
    color: @text-color;
  }

  &:focus, &:active {
    background-color: transparent;
    .focused-btn();
  }
}

.ant-btn-sm.custom-btn {
  padding: @padding-minimal @padding-tiny;
  font-size: @font-size-very-small;
  border-radius: @border-radius-small;
}