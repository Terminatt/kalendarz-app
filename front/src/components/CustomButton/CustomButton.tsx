import { Button, ButtonProps } from 'antd';
import React from 'react';

import './CustomButton.less';

export type CustomButtonVariant = 'primary' | 'clear';

export interface CustomButtonProps extends ButtonProps {
  variant: CustomButtonVariant;
}

const CustomButton: React.FC<CustomButtonProps> = (props) => {
    const { variant, children, ...rest } = props;

    return <Button className={`custom-btn custom-btn-${variant}`} {...rest}>{children}</Button>;
};

export default CustomButton;
