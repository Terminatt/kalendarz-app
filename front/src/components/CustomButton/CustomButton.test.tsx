import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from './CustomButton';
import { MenuOutlined } from '@ant-design/icons';
import { render } from 'react-dom';

describe('Custom Button Component Variants', () => {
    it('Primary matches the snapshot', () => {
        const tree = renderer
            .create(
                <CustomButton variant="primary">Test</CustomButton>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('Default matches the snapshot', () => {
        const tree = renderer
            .create(
                <CustomButton variant="default">Test</CustomButton>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('Delete matches the snapshot', () => {
        const tree = renderer
            .create(
                <CustomButton variant="delete">Test</CustomButton>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('Clear matches the snapshot', () => {
        const tree = renderer
            .create(
                <CustomButton variant="clear">Test</CustomButton>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('Icon matches the snapshot', () => {
        const tree = renderer
            .create(
                <CustomButton variant="icon"><MenuOutlined /></CustomButton>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });


    it('Icon Border matches the snapshot', () => {
        const tree = renderer
            .create(
                <CustomButton variant="icon-border"><MenuOutlined /></CustomButton>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('Minimal matches the snapshot', () => {
        const tree = renderer
            .create(
                <CustomButton variant="minimal">Test</CustomButton>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });
});

describe('Custom Button component', () => {
    it('should render text', () => {
        const container = document.createElement('div');
        const text = 'Test'

        render(
            <CustomButton>
                {text}
            </CustomButton>, container
        )
        
        expect(container.firstChild?.textContent).toBe(text);
    })
})
