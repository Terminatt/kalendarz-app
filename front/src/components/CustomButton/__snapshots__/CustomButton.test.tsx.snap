// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Button Component Variants Clear matches the snapshot 1`] = `
<button
  className="ant-btn custom-btn custom-btn-clear"
  onClick={[Function]}
  type="button"
>
  <span>
    Test
  </span>
</button>
`;

exports[`Custom Button Component Variants Default matches the snapshot 1`] = `
<button
  className="ant-btn custom-btn custom-btn-default"
  onClick={[Function]}
  type="button"
>
  <span>
    Test
  </span>
</button>
`;

exports[`Custom Button Component Variants Delete matches the snapshot 1`] = `
<button
  className="ant-btn custom-btn custom-btn-delete"
  onClick={[Function]}
  type="button"
>
  <span>
    Test
  </span>
</button>
`;

exports[`Custom Button Component Variants Icon Border matches the snapshot 1`] = `
<button
  className="ant-btn custom-btn custom-btn-icon-border"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="menu"
    className="anticon anticon-menu"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="menu"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
      />
    </svg>
  </span>
</button>
`;

exports[`Custom Button Component Variants Icon matches the snapshot 1`] = `
<button
  className="ant-btn custom-btn custom-btn-icon"
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="menu"
    className="anticon anticon-menu"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="menu"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
      />
    </svg>
  </span>
</button>
`;

exports[`Custom Button Component Variants Minimal matches the snapshot 1`] = `
<button
  className="ant-btn custom-btn custom-btn-minimal"
  onClick={[Function]}
  type="button"
>
  <span>
    Test
  </span>
</button>
`;

exports[`Custom Button Component Variants Primary matches the snapshot 1`] = `
<button
  className="ant-btn custom-btn custom-btn-primary"
  onClick={[Function]}
  type="button"
>
  <span>
    Test
  </span>
</button>
`;
